### PR TITLE
Update Hoa dependency

### PR DESCRIPTION
**Pull request by jubianchi**
Thanks @Hywan for the work he did on Hoa itself and for the initial PR (#29)

### CODE REVIEW

**Comment by jubianchi** [comment]
OK, there is a tiny bug in atoum autoloader: it does not handle traits. Let me fix it.

**Comment by jubianchi** [comment]
everything is in atoum/atoum#556.

**Comment by Hywan** [comment]
I have restarted the build.

**Comment by Hywan** [comment]
Hmm, it does not work. @jubianchi can you restart the build?

**Comment by jubianchi** [comment]
@Hywan let me check ;)

**Comment by Hywan** [comment]
Is it working?

**Comment by jubianchi** [comment]
@Hywan not really, it still fails with `--prefer-lowest`...

**Comment by agallou** [comment]
@jubianchi @Hywan some minimum version on hoa/ruler dependancies are missing (https://github.com/hoaproject/Central/issues/31). So we had to add them to atoum/ruler-extension (https://github.com/atoum/ruler-extension/pull/22). Those are not present on the PR, so the prefe-lowest test fails. Maybe we should keep them.

File: autoloader.php [hunk 1]
```diff
@@ -11,21 +11,72 @@
 	$vendorDirectory = __DIR__ . '/../..';
 }
 
-if (is_file($vendorAutoloader = $vendorDirectory . '/hoa/core/Core.php')) {
-	require_once $vendorDirectory . '/hoa/core/Core.php';
+$hoa = array(
```
Comment (Hywan) [inline autoloader.php:14]: This is fragile. Imagine we are adding a new dependency. This mapping will break. Consider that if something is null present, it acts as `null`.

**Pull request merged**